### ISSUE TITLE
Spin off configuration schema generator as a standalone package

### ISSUE THREAD

**Issue by pedropaulovc** [dotnet/aspire#3309]
Autocomplete of Aspire configs in appsettings.json is fantastic as my project is growing. I would love to have this capability in my projects as well. Also, I'm using keyed component resolution and the default Aspire schema doesn't know what's the key I'm using in my code. Looks like this capability is handled by the configuration schema generator introduced in #1383 . The team should consider spinning off this capability as an extension package for 3P usage.

**Comment by davidfowl** [comment]
This is the plan right @eerhardt?

**Comment by eerhardt** [comment]
> This is the plan right @eerhardt?

Yes, we will need to do something here if we want to build Aspire Components outside of the dotnet/aspire repo.

**Comment by aaronpowell** [comment]
We're wanting to use this in the Community Toolkit (tracking https://github.com/CommunityToolkit/Aspire/issues/112), what would be required to ship this as a .NET CLI tool that we could use, even if it was on a package feed other than nuget.org.

**Comment by Alirexaa** [comment]
@davidfowl @eerhardt
Is there any plan for this in the near future?
Maybe an aspire cli command?

I would like to resolve this issue if you can provide some information about how this should be designed.

**Comment by maddymontaquila** [comment]
The configuration schema generator has been spun off and improved via PRs #4441, #1318, and #6312. Closing as implemented.

**Comment by KalleOlaviNiemitalo** [comment]
> The configuration schema generator has been spun off and improved via PRs #4441, #1318, and #6312. Closing as implemented.

#1318 is not a pull request and does not look related.

What is the name of the package?  It is not clear to me from <https://github.com/microsoft/aspire/blob/5c8aa3df3eba94f3d71f7ab4b276f03e8233955e/src/Tools/ConfigurationSchemaGenerator/ConfigurationSchemaGenerator.csproj> how this is being packaged.

**Comment by davidfowl** [comment]
@eerhardt are we gonna do this still? It's not really a priorty anymore is it.

**Comment by Meir017** [comment]
could this tool move into https://github.com/dotnet/extensions ?

**Comment by bart-vmware** [comment]
Steeltoe still has an internal copy of this tool. It would be great if we could just reference a NuGet package instead.

**Comment by davidfowl** [comment]
This just isn't a priority for the team. If someone is inspired to move it to extesnions and get it merged that would be great.
